students/ericstreit/session03/mailroom.py

```diff
@@ -32,45 +32,54 @@ def mainmenu():
             print("Goodbye!")
             break
         else:
-             print("Please enter 'S', 'L' or 'Q'")
+             print("\nPlease enter 'S', 'R' or 'Q'")
 
 def thankyou():
-    """This is the thank you menu"""
-    name = input("Please enter the name of the donor or type 'list' to list current donors.")
+    """This is the thank you menu that routes to the listuser or donor_update function"""
+    print("Please enter the name of the donor or type 'list' to list current donors.")
+    name = input(": ")
     if name == "list":
-        listusers()
+        list_users()
     else:
-        donorupdate(name)
+        donor_update(name)
+
+def list_users():
+    """A simple function that lists donor names"""
+    print("\nDonors\n", "{:-<60}".format(""))
+    for i in donors:
+        print(i[0])
 
 def compose_email(name, amount):
     """This function will compose the thank you email"""
     print("\n\nEMAIL SENT:\n'Thank you, {}, for your generous contribution of ${}! We will be sure to ask you for more money again soon.'\n\n".format(name, amount))
 
-def donorupdate(name):
+def donor_update(name):
     """This function will update the donor list with a new donor or a new amount given by a current donor"""
-    l = len(donors)
-    print(name)
+#    l = len(donors)
+#    print(name)
+#variable if a name is found in the search. default is 0 / False
     found = 0
+#for loop to search for a donor name in the database. If found it will flag the found variable as True and add the addtional amount
     for i in range(len(donors)):
         if name in donors[i]:
-            amount = int(input("How much $$ did they donate this time?: "))
+            print("How much $$ did they donate this time?: ")
+            amount = int(input(": "))
             donors[i][1].append(amount)
             found = 1
             break
         else:
             continue
     if not found:
```

Comment:
you can use a "else" block with the for loop instead of this "sentinel" value?

https://uwpce-pythoncert.github.io/PythonCertDevel/modules/Iteration.html#else